packages/components/src/__actions__/Button/v3/_docs/Button--api-specification.mdx

```diff
@@ -0,0 +1,160 @@
+
+
+### Button content and children
+
+Labels and any button content can be passed into the `Button` as `children`. Content wrapped by the `Button` will be spaced using `flex-gap` relative to the `size` prop. This enables consistent spacing in `RTL` layouts.
+
+```tsx
+<Button variant="secondary" onPress={e => submit(e)}>
+  Label
+</Button>
+```
+
+While in most cases, `children` will be a `ReactNode`, `Button` also accepts a render function with React Aria's `ButtonRenderProps`. This allows for more advanced styling and rendering options by hooking into React Aria's internal button state. You can read more about this [here](https://react-spectrum.adobe.com/react-aria/Button.html#styling).
+
+
+### Icons and positioning
+
+The `icon` property abstracts the need to handle positioning and sizing logic for icons within the `Button`. When paired with the [Icon component](/docs/illustrations-icon-icon-future-api-specification--docs), this will scale optical size and stroke width `size` prop.
+
+<Canvas of={exampleStories.ButtonWithIcon} />
+
+Set the position of the icon using the `iconPosition` prop. This will ensure content is flipped in `RTL` layouts.
+
+<Canvas of={exampleStories.ButtonWithIconEnd} />
+
+Note that icons will need the [shouldMirrorInRTL](/docs/illustrations-icon-icon-future-api-specification--docs#mirror-in-rtl) prop set to `true` when mirroring is required.
+
+
+### Icon only buttons
+
+The recommended pattern to achieve an icon only button (aka: `IconButton`) is to pass an icon in as props, then to use a [VisuallyHidden](/docs/components-visually-hidden--docs) component as `children`.
+
+<Canvas of={exampleStories.IconButton} className="mb-32"  />
+
+This pattern ensures that the `Button`'s accessible name is determined by its children, which helps to announce relevant content to the screen readers, as with the [pending state](#pending-state). You can learn more about this [accessible pattern here](https://cultureamp.atlassian.net/wiki/spaces/PA/pages/3833331831/Accessible+button+and+link+labels).
+
+### Pending state
+
+`isPending` and `pendingLabel` are used to indicate that an interaction is in progress. This will disable the `Button`'s press events while rendering a loading spinner and `pendingLabel`.
+
+<Canvas of={specStories.PendingButton} />
+
+For an icon only `Button`, this can be paired with the `isPendingLabelHidden` prop to visibly hide the `pendingLabel` to avoid layout shifts. It is important that the `pendingLabel` is still given meaningful message as this will be announced to a screen reader.
+
+### Full width Buttons
+
+If a button is statically the full width of a container you can use `isFullWidth` property.
+
+<Canvas of={exampleStories.ButtonFullWidth} />
+
+For resizing on on smaller screens, consider using the `className` to leverage CSS media or container queries, ie: `<Button className="w-full md:w-[initial]">Label</Button>`.
+
+
+### Styling hover, focus and other pseudo-classes
```

Comment:
Any chance we don't advertise this at all? We don't want to encourage people to change styles, and on the rare occasion someone does have that edge case, both options probably work anyway.

Reply:
I suppose the issue is there's no way of them knowing this without digging into the react aria docs. I'd be hesitant in removing in case since assumed behavior would be that the component would uses standard pseudo classes.

Reply:
That said it still works if they pass it into className, I just honestly don't know, which one will win out in specificity. Maybe I'm think too much towards an edge case?

Reply:
I think you're just over-thinking it :) Wait until it proves to actually be a problem first - by that time you may even want to pivot your solution (definitely done that before myself)

Reply:
Fair point, will axe the section 👍